omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_9/SystemSettingResource1_9.java

```diff
@@ -0,0 +1,255 @@
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link Resource} for {@link GlobalProperty}, supporting standard CRUD
+ * operations
+ */
+@Resource(name = RestConstants.VERSION_1 + "/systemsetting", supportedClass = GlobalProperty.class, supportedOpenmrsVersions = {"1.9.*", "1.10.*", "1.11.*"})
+public class SystemSettingResource1_9 extends DelegatingCrudResource<GlobalProperty> {
+
+    public static final String GENERAL = "General Settings";
+
+    @RepHandler(RefRepresentation.class)
+    public SimpleObject asRef(GlobalProperty delegate) throws ConversionException {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("uuid");
+        description.addProperty("property");
+        description.addProperty("propertyValue");
+        description.addProperty("description");
+        description.addProperty("display", findMethod("getDisplayName"));
+        description.addSelfLink();
+        return convertDelegateToRepresentation(delegate, description);
+    }
+
+    @RepHandler(FullRepresentation.class)
+    public SimpleObject asFull(GlobalProperty delegate) throws ConversionException {
+        DelegatingResourceDescription description = fullRepresentationDescription(delegate);
+        return convertDelegateToRepresentation(delegate, description);
+    }
+
+    protected DelegatingResourceDescription fullRepresentationDescription(GlobalProperty delegate) {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("uuid");
+        description.addProperty("property");
+        description.addProperty("propertyValue");
+        description.addProperty("description");
+        description.addProperty("display", findMethod("getDisplayName"));
+        description.addProperty("datatypeClassname");
+        description.addProperty("datatypeConfig");
+        description.addProperty("preferredHandlerClassname");
+        description.addProperty("handlerConfig");
+        description.addSelfLink();
+        return description;
+    }
+
+    /**
+     * @see DelegatingCrudResource#getRepresentationDescription(Representation)
+     */
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
```

Comment:
Why use RepHandler for two of the representations, and getRepresentationDescription for the third? I would suggest you just do if (ref) ... else if (default) ... else if (full) ... in this method